smartdashboard/views.py

```diff
@@ -341,3 +343,448 @@ def __init__(
+            try:
+                delta_df = pd.read_csv(self.files[0], skiprows=range(1, skiprows))
+                return delta_df
+            except FileNotFoundError:
+                self.table_element.info(self.message)
+                return pd.DataFrame()
+        return pd.DataFrame(columns=self.columns)
+
+    def _get_data_file(self) -> str:
+        """On click event to return csv data for the export button
+
+        :return: CSV data
+        :rtype: str
+        """
+        if self.shard is not None:
+            try:
+                return pd.read_csv(self.files[0]).to_csv()
+            except FileNotFoundError:
+                self.table_element.info(self.message)
+                self.telemetry = False
+        return ""
+
+    def _load_data(self) -> pd.DataFrame:
+        """Load initial data
+
+        :return: Initial dataframe
+        :rtype: pandas.DataFrame
+        """
+        if self.shard is not None and self.telemetry:
+            telem_df = pd.read_csv(self.files[0])
+            return telem_df
+
+        return pd.DataFrame(columns=self.columns)
+
+
+class MemoryView(DualView):
+    """View class for memory section of the Database Telemetry page"""
+
+    def enable_export_button(self) -> None:
+        """Create an export data button"""
+        if self.shard is not None and self.telemetry:
+            try:
+                self.export_button.download_button(
+                    label="Export Data",
+                    data=self._get_data_file(),
+                    file_name=f"{self.shard.name} memory.csv",
+                    mime="text/csv",
+                    key="memory",
+                )
+            except FileNotFoundError:
+                self.export_button.empty()
+        else:
+            self.export_button.empty()
+
+    def _get_files(self) -> t.Tuple[str, str]:
+        """Returns a tuple of the files used for telemetry"""
+        if self.shard is not None:
+            return self.shard.memory_file, self.shard.memory_file
+
+        return ("", "")
+
+    def _get_message(self) -> str:
+        """Returns the error message if files cannot be found"""
+        if self.shard is not None:
+            return f"Memory information could not be found for {self.shard.name}"
+
+        return ""
+
+    def get_columns(self) -> t.List[str]:
+        """Returns columns for the graph dataframe"""
+        return ["timestamp", "used_memory", "used_memory_peak", "total_system_memory"]
+
+    def initial_load(self) -> None:
+        """Updates the table and graph with initial dataframes"""
+        table_df = self.telemetry_df.copy(deep=True)
+        graph_df: pd.DataFrame = self.telemetry_df.copy(deep=True)
+        if self.telemetry_df.copy(deep=True).shape[0] >= self.window_size:
+            graph_df = graph_df.sample(self.window_size)
+
+        self._update_graph(self.process_dataframe(graph_df))
+        self._update_table(self.process_dataframe(table_df))
```

Comment:
Minor "performance" thing: You can save an extra `process_dataframe` call, if you do a little reorg here:
```suggestion
    def initial_load(self) -> None:
        """Updates the table and graph with initial dataframes"""
        table_df = self.telemetry_df.copy(deep=True)
        table_df = self.process_dataframe(table_tf)
        graph_df: pd.DataFrame = table_tf.copy(deep=True)
        if self.telemetry_df.copy(deep=True).shape[0] >= self.window_size:
            graph_df = graph_df.sample(self.window_size)

        self._update_graph(graph_df)
        self._update_table(table_df)
```

Reply:
Love it!